runtime/src/qio/qio_popen.c

```diff
@@ -384,16 +384,26 @@ qioerr qio_proc_communicate(
 
   if( threadsafe ) {
     // lock all three channels.
+    // but unlock them immediately and set them to NULL
+    // if they are already closed.
```

Comment:
Setting it to NULL affects only the local variable and doesn't actually change the channel's data. It just means the rest of this function won't use that channel.